seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java

```diff
@@ -132,6 +133,9 @@ private int getBytesForValue(Object v, SeaTunnelDataType<?> dataType) {
         SqlType sqlType = dataType.getSqlType();
         switch (sqlType) {
             case STRING:
+                if (v instanceof BigDecimal) {
+                    return v.toString().length();
+                }
```

Comment:
If it is declared as string, but the returned type is BigDecimal, this is wrong. We need to solve this wrong behavior instead of condoning it.